tests/udf_communication/peer_communication/test_send_recv.py

```diff
@@ -0,0 +1,77 @@
+import random
+import time
+from pathlib import Path
+from typing import Dict, Set, List
+
+import structlog
+from structlog import WriteLoggerFactory
+
+from exasol_advanced_analytics_framework.udf_communication.connection_info import ConnectionInfo
+from exasol_advanced_analytics_framework.udf_communication.ip_address import IPAddress
+from exasol_advanced_analytics_framework.udf_communication.peer_communicator import PeerCommunicator
+from tests.udf_communication.peer_communication.utils import TestProcess, BidirectionalQueue, assert_processes_finish
+
+structlog.configure(
+    context_class=dict,
+    logger_factory=WriteLoggerFactory(file=Path(__file__).with_suffix(".log").open("wt")),
+    processors=[
+        structlog.contextvars.merge_contextvars,
+        structlog.processors.add_log_level,
+        structlog.processors.StackInfoRenderer(),
+        structlog.dev.set_exc_info,
+        structlog.processors.TimeStamper(),
+        structlog.processors.JSONRenderer()
+    ]
+)
+
+
+def run(name: str, group_identifier: str, number_of_instances: int, queue: BidirectionalQueue):
+    listen_ip = IPAddress(ip_address=f"127.1.0.1")
+    com = PeerCommunicator(
+        name=name,
+        number_of_peers=number_of_instances,
+        listen_ip=listen_ip,
+        group_identifier=group_identifier)
+    queue.put(com.my_connection_info)
+    peer_connection_infos = queue.get()
+    for index, connection_infos in peer_connection_infos.items():
+        com.register_peer(connection_infos)
+        time.sleep(random.random() / 10)
```

Comment:
add comment that this make the process sleep a random amount of time between 0 and 100ms

Reply:
removed it, makes the test to slow, this likely get tested anyway by the test_local_discovery.py. Else, we could also add a specific test for this.